commit fbe15b7184da09c0d71051bf3978540f63aba5cc
Author: Olivier Houchard <cognet@ci0.org>
Date:   Sun Jan 5 16:45:14 2020 +0100

    MINOR: ssl: Remove unused variable "need_out".
    
    The "need_out" variable was used to let the ssl code know we're done
    reading early data, and we should start the handshake.
    Now that the handshake function is responsible for taking care of reading
    early data, all that logic has been removed from ssl_sock_to_buf(), but
    need_out was forgotten, and left. Remove it know.
    This patch was submitted by William Dauchy <w.dauchy@criteo.com>, and should
    fix github issue #434.
    This should be backported to 2.0 and 2.1.
    
    (cherry picked from commit 7f4f7f140f6b03b61d1b38260962db235c42c121)
    Signed-off-by: Christopher Faulet <cfaulet@haproxy.com>

diff --git a/src/ssl_sock.c b/src/ssl_sock.c
index 7c62299a0..1fac2d905 100644
--- a/src/ssl_sock.c
+++ b/src/ssl_sock.c
@@ -6385,7 +6385,6 @@ static size_t ssl_sock_to_buf(struct connection *conn, void *xprt_ctx, struct bu
 	 * EINTR too.
 	 */
 	while (count > 0) {
-		int need_out = 0;
 
 		try = b_contig_space(buf);
 		if (!try)
@@ -6443,8 +6442,6 @@ static size_t ssl_sock_to_buf(struct connection *conn, void *xprt_ctx, struct bu
 			/* otherwise it's a real error */
 			goto out_error;
 		}
-		if (need_out)
-			break;
 	}
  leave:
 	return done;
